task_02/danil_dyachenko/02_knight_and_princess/graph_gen.cpp

```diff
@@ -0,0 +1,107 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+struct Vertex{
+    int id;
+    std::vector<int> edges;
+};
+
+struct Edge{
+    int id;
+    std::vector<int> vertices;
+};
+
+class Graph{
+    std::vector<Vertex> vertices;
+    std::vector<Edge> edges;
+public:
+    void add_vertices(int n_vertices = 1){
+        int n_edges = 1;
+        int x;
+        vertices.resize(n_vertices);
+        for (int i = 0; i < n_vertices; ++i){
+            std::cout << "n_edges for " << i << " vertex: ";
+            std::cin >> n_edges;
+            vertices[i].edges.resize(n_edges);
+            vertices[i].id = i;
+            for (int j = 0; j < n_edges; ++j){
+                std::cin >> x;
+                vertices[i].edges[j] = x;
+            }
+        }
+    }
+    void print_vertices(void){
+        std::cout << "Printing vertices:" << std::endl;
+        for (int i = 0; i < vertices.size(); ++i){
+            std::cout << i << ": [ ";
+            for (int j = 0; j < vertices[i].edges.size(); ++j)
+                std::cout << vertices[i].edges[j] << " ";
+            std::cout << "]" << std::endl;
+        }   
+    }
+    void add_edges(int n_edges = 1){
+        int n_vertices = 2;
+        int x, y;
+        edges.resize(n_edges);
+        for (int i = 0; i < n_edges; ++i){
+            edges[i].vertices.resize(n_vertices);
+            edges[i].id = i;
+            std::cout << i << " edge: " << std::endl;
+            for (int j = 0; j < n_vertices; ++j){
+                std::cin >> x;
+                edges[i].vertices[j] = x;
+            }
+        }
+    }
+    void print_edges(void){
+        std::cout << "Printing edges:" << std::endl;
+        for (int i = 0; i < edges.size(); ++i){
+            std::cout << i << ": [ ";
+            for (int j = 0; j < edges[i].vertices.size(); ++j)
+                std::cout << edges[i].vertices[j] << " ";
+            std::cout << "]" << std::endl;
+        }   
+    }
+    void to_json(const char * name){
```

Comment:
В `C++` есть специальная структура данных для работы со строками
```suggestion
    void to_json(const std::string& name){
```

Reply:
Это не зона отвественности графа - работа с файлами. Скорее данная функция должна возвращать строку:
```suggestion
    std::string to_json() const {
```
И полученная строка уже записывается в файл в другой части программы.